tests/unit/mathematics/linalg/operations/Eigen3_operations_unittest.cc

```diff
@@ -189,6 +190,43 @@ TEST(LinalgBackendEigen, SGMatrix_add_col_vec_in_place)
 		}
 }
 
+TEST(LinalgBackendEigen, add_diag)
+{
+	SGMatrix<float64_t> A(2, 3);
+	SGVector<float64_t> b(2);
+
+	A(0, 0) = 1;
+	A(0, 1) = 2;
+	A(0, 2) = 3;
+	A(1, 0) = 4;
+	A(1, 1) = 5;
+	A(1, 2) = 6;
+
+	b[0] = 1;
+	b[1] = 2;
+
+	add_diag(A, b, 0.5, 2.0);
+
+	EXPECT_NEAR(A(0, 0), 2.5, 1e-15);
+	EXPECT_NEAR(A(0, 1), 2, 1e-15);
+	EXPECT_NEAR(A(0, 2), 3, 1e-15);
+	EXPECT_NEAR(A(1, 0), 4, 1e-15);
+	EXPECT_NEAR(A(1, 1), 6.5, 1e-15);
+	EXPECT_NEAR(A(1, 2), 6, 1e-15);
+}
+
+TEST(LinalgBackendEigen, add_diag_error)
+{
+	SGMatrix<float64_t> A1(2, 2);
+	SGVector<float64_t> b1(3);
+	SGMatrix<float64_t> A2;
+	SGVector<float64_t> b2;
+	EXPECT_THROW(add_diag(A1, b1), ShogunException);
+	EXPECT_THROW(add_diag(A1, b2), ShogunException);
+	EXPECT_THROW(add_diag(A2, b1), ShogunException);
+	EXPECT_THROW(add_diag(A2, b2), ShogunException);
+}
```

Comment:
IMHO, I'd shove all `add_diag`-related tests into one test unit for the neatness.
Otherwise good to me :)